### PR TITLE
castbar: Update safezone position according to the castbar orientation

### PR DESCRIPTION
**Pull request by Rainrider**
Fixes https://github.com/oUF-wow/oUF/issues/526

Dry-coded

### CODE REVIEW

**Comment by LybrialsGit** [comment]
For channeling the safe zone needs to be inverted, I guess:

- casting, horizontal, not reverse fill: SafeZone right
- casting, horizontal, reverse fill: SafeZone left
- casting, vertical, not reverse fill: SafeZone top
- casting, vertical, reverse fill: SafeZone bottom

- channeling, horizontal, not reverse fill: SafeZone left
- channeling, horizontal, reverse fill: SafeZone right
- channeling, vertical, not reverse fill: SafeZone bottom
- channeling, vertical, reverse fill: SafeZone top

What do you think?

**Comment by Rainrider** [comment]
Ah yes, it was reverted before as well. Will fix a bit later. Thanks for the feedback!

File: elements/castbar.lua [hunk 1]
```diff
@@ -107,16 +107,26 @@ local function updateSafeZone(self)
...
+		)
+	else
+		sz:SetPoint(
+			self:GetReverseFill() and
+			(horiz and 'RIGHT' or 'TOP') or
+			(horiz and 'LEFT' or 'BOTTOM')
+		)
+	end
+
 	sz:SetPoint(horiz and 'TOP' or 'LEFT')
```
Comment (LybrialsGit) [inline elements/castbar.lua:130]: Do you need to do the differentiation between channeling and casting here too?

**Comment by LybrialsGit** [comment]
This looks good now.

File: elements/castbar.lua [hunk 2]
```diff
@@ -142,13 +167,9 @@ local function UNIT_SPELLCAST_START(self, event, unit)
...
-	local sf = element.SafeZone
-	if(sf) then
-		sf:ClearAllPoints()
-		sf:SetPoint(element:GetReverseFill() and 'LEFT' or 'RIGHT')
-		sf:SetPoint('TOP')
-		sf:SetPoint('BOTTOM')
-		updateSafeZone(element)
+	if (element.SafeZone) then
+		positionSafeZone(element);
+		updateSafeZone(element);
```
Comment (p3lim) [inline elements/castbar.lua:172]: Too much PHP/JS lately? :P

Comment (Rainrider) [reply elements/castbar.lua:172]: Heh? :)

Comment (Rainrider) [reply elements/castbar.lua:172]: Hahahah you got me with those semi-colons

**Comment by Rainrider** [comment]
@LybrialsGit you tested this in-game right?

**Comment by LybrialsGit** [comment]
Looks fine ingame

**Comment by Rainrider** [comment]
horiz
horiz-reverse
vertical
vertical-reverse

LGTM

**Comment by LybrialsGit** [comment]
Looks great. So you are not a fan of spacec behind control structures? And you also seem to discriminate semicolons :D
After I will release my self coded interface you wont like my code at all. My first programming language 15 years ago was Java. So all my code is full of whitespaces and semicolons :D

**Comment by Rainrider** [comment]
I do like spaces after control structures reserved words to differentiate them from functions but one could argue about the sense of that since there are only a few of those. Semicolons are mostly noise in Lua. We have democratically decided to follow a [styleguide in oUF](https://github.com/oUF-wow/oUF/wiki/Styleguide) so I try to stick to that, because code consistency is a huge time saver when working in a team. But well, 5 out of 7 commits just to fix style issues show clearly I haven't done any Lua in a long time :)

And yes, Java has awful styleguides, at least those by Sun and Google are not to my taste :)

**Pull request merged**